File: frontend/src/models/Expense.ts
export type ExpenseCategory =
  | "Transport"
  | "Accommodation"
  | "Food"
  | "Tickets"
  | "Shopping"
  | "Other";

export interface Expense {
  id: string;
  tripPlanId: string;
  name: string;
  category: ExpenseCategory;
  amount: number;
  date: string;
  description: string;
  createdAt: string;
}

export interface ExpenseRequest {
  tripPlanId: string;
  name: string;
  category: ExpenseCategory;
  amount: number;
  date: string;
  description: string;
}

export interface BudgetSummary {
  tripPlanId: string;
  plannedBudget: number;
  totalSpent: number;
  remainingBudget: number;
}

export const EXPENSE_CATEGORY_LABELS: Record<ExpenseCategory, string> = {
  Transport: "Prevoz",
  Accommodation: "Smeštaj",
  Food: "Hrana",
  Tickets: "Ulaznice",
  Shopping: "Kupovina",
  Other: "Ostalo",
};

File: backend/ExpenseService/Models/Expense.cs
using System;

namespace ExpenseService.Models
{
    public class Expense
    {
        public Guid Id { get; set; }

        public Guid TripPlanId { get; set; }

        public Guid UserId { get; set; }

        public string Name { get; set; }

        public ExpenseCategory Category { get; set; }

        public decimal Amount { get; set; }

        public DateTime Date { get; set; }

        public string Description { get; set; }

        public DateTime CreatedAt { get; set; }
    }
}
